exporters/fluentd/src/log/recordable.cc

```diff
@@ -64,6 +72,18 @@ void Recordable::SetTimestamp(
           1000000000);
 }
 
+void Recordable::SetObservedTimestamp(
+    opentelemetry::common::SystemTimestamp timestamp) noexcept {
+  json_["ObservedTimestamp"] = fluentd_common::get_msgpack_eventtimeext(
+      static_cast<int32_t>(std::chrono::duration_cast<std::chrono::seconds>(
+                               timestamp.time_since_epoch())
+                               .count()),
+      std::chrono::duration_cast<std::chrono::nanoseconds>(
+          timestamp.time_since_epoch())
+              .count() %
+          1000000000);
+}
```

Comment:
should we move this code to some common function to be used in both SetTimestamp and SetObservedTimestamp ?